### PR TITLE
fix: make FFmpegOpusAudio codec behave as docs suggest

### PR DESCRIPTION
**Pull request by felix920506**
## Summary

make FFmpegOpusAudio "codec" parameter behave as what the inline docs suggest

Original behavior:
passing in "libopus" or "opus" will result in "copy" being passed into ffmpeg
passing in anything else (including "copy") will result in "libopus" being passed into ffmpeg

New behavior:
passing in "copy" will result in "copy" being passed into ffmpeg
passing in anything else (including "libopus", "opus") will result in "libopus" being passed into ffmpeg

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
- [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by plun1331** [comment]
should we maybe change the docs instead because like, breaking changes and stuff

**Comment by felix920506** [comment]
This is my first contribution here so I don't really know if this would count as breaking or not. I checked the box only because it technically changes the behavior of this function.

**Comment by felix920506** [comment]
I don't think this is going to be much an issue since the only scenario it would break functionality on existing bots would be someone passing in "copy" for a source that isn't opus, which is something that was never supposed to work according to the docs. 
The fix this will require under that scenario would be a few lines at most (Likely single line changes for most cases.)

**Comment by felix920506** [comment]
I see this tagged with "changelog needed", is there anything that I need to do? If so are there instructions?

**Comment by JustaSqu1d** [comment]
> I see this tagged with "changelog needed", is there anything that I need to do? If so are there instructions?

Here's an example of a changelog entry being added: https://github.com/Pycord-Development/pycord/pull/2577/commits/7fb1a4b07c1f4b2149a473f5287c0df2fa877474

**Comment by felix920506** [comment]
> > I see this tagged with "changelog needed", is there anything that I need to do? If so are there instructions?
> 
> 
> 
> Here's an example of a changelog entry being added: https://github.com/Pycord-Development/pycord/pull/2577/commits/7fb1a4b07c1f4b2149a473f5287c0df2fa877474

would this be "Fixed" or "Changed" ?

**Comment by Lulalaby** [comment]
Fixed

**Comment by felix920506** [comment]
I have added an entry to the changelog.

File: discord/player.py [hunk 1]
```diff
@@ -501,6 +511,8 @@ def custom_probe(source, executable):
 
         executable = kwargs.get("executable")
         codec, bitrate = await cls.probe(source, method=method, executable=executable)
+        # only re-encode ir source isn't already opus, else directly copy source audio stream
```
Comment (JustaSqu1d) [inline discord/player.py:514]:
```suggestion
        # only re-encode if the source isn't already opus, else directly copy the source audio stream
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Pull request merged**